pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/SourceDistTask.java

```diff
@@ -22,14 +22,14 @@
 import com.linkedin.gradle.python.tasks.execution.TeeOutputContainer;
 import com.linkedin.gradle.python.util.VirtualEnvExecutableHelper;
 import org.gradle.api.Action;
-import org.gradle.api.DefaultTask;
 import org.gradle.api.Project;
 import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.process.ExecResult;
 import org.gradle.process.ExecSpec;
 
 
-public class SourceDistTask extends DefaultTask implements FailureReasonProvider {
+public class SourceDistTask extends AbstractPythonMainSourceDefaultTask implements FailureReasonProvider {
 
     private TeeOutputContainer container = new TeeOutputContainer();
     @TaskAction
```

Comment:
If we're switching to use AbstractPythonMainSourceDefaultTask let's use it all the way.

We can get rid of the `packageSdist()` method and create a public constructor that adds the args like we do in `com.linkedin.gradle.python.tasks.Flake8Task`